runtime/list.go

```diff
@@ -170,6 +170,38 @@ func listCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return seqCount(f, args[0], args[1])
 }
 
+func listRemove(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "remove", args, ListType, ObjectType); raised != nil {
+		return nil, raised
+	}
+	v := args[1]
+	i := -1
+	pred := func(o *Object) (bool, *BaseException) {
+		i++
+		eq, raised := Eq(f, o, v)
+		if raised != nil {
+			return false, raised
+		}
+		ret, raised := IsTrue(f, eq)
+		if raised != nil {
+			return false, raised
+		}
+		return ret, nil
+	}
+	l := toListUnsafe(args[0])
+	l.mutex.RLock()
```

Comment:
Recursive mutex is a [contentious topic](https://groups.google.com/forum/#!topic/golang-nuts/vN5ncBdtkcA) in the Go community. It was intentionally left out because they can be error prone.

As it happens, it's not possible to implement some Python functionality without recursive locks, so I've had to [implement](https://github.com/google/grumpy/blob/master/runtime/threading.go#L49) a cheesy recursive lock for Grumpy. This is likely also how we'll implement the Python threading.RLock class.